docs/secret_management.md

```diff
@@ -169,9 +170,57 @@ spec:
         value: "/my-secret-backend.sh"
 ```
 
+As in the Datadog Operator, the Datadog Agent image includes a helper function `readsecret.sh` that can be used to read mounted secrets. After creating the secret and setting the volume mount (in any container that requires it), set the `DD_SECRET_BACKEND_COMMAND` and `DD_SECRET_BACKEND_ARGUMENTS` environmental variables.
+
+For instance, to use the secret backend for the Agent and Cluster Agent, create a secret called "test-secret":
+
+`kubectl create secret generic test-secret --from-literal=api_key='<api-key>' --from-literal=app_key='<app-key>'`
+
+And then set the DatadogAgent spec:
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog
+spec:
+  credentials:
+    apiKey: ENC[api_key]
+    appKey: ENC[app_key]
+    useSecretBackend: true
+  agent:
+    env:
+      - name: DD_SECRET_BACKEND_COMMAND
+        value: "/readsecret.sh"
+      - name: DD_SECRET_BACKEND_ARGUMENTS
+        value: "/etc/secret-volume"
```

Comment:
For `v2.DatadogAgent` we could imaging to have specific fields for configuring the secret-backend

Reply:
i have a list of a couple changes to make for v2 regarding this feature, will add this too